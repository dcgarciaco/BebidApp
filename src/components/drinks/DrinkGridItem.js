import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export const DrinkGridItem = ({ drinkID, drinkName, drinkImageURL }) => {
  return (
    <Link
      to={`./drink/${drinkID}`}
      className='card animate__animated animate__fadeIn'
    >
      <img src={drinkImageURL} alt={drinkName} />
      <p>{drinkName}</p>
    </Link>
  );
};

DrinkGridItem.propTypes = {
  drinkID: PropTypes.string.isRequired,
  drinkName: PropTypes.string.isRequired,
  drinkImageURL: PropTypes.string.isRequired,
};
