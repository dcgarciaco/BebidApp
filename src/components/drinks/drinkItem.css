.drinkContent {
  display: grid;
  grid-template-columns: 45% 2fr;
  grid-template-rows: 1fr;
  gap: 0px 0px;
  grid-auto-flow: row;
  grid-template-areas: '. .';
  color: #002d29;
}

figure {
  overflow: hidden;
  max-width: 350px;
  justify-self: end;
  align-self: center;
}

.drinkContent img {
  max-width: 100%;
  border-radius: 6px;
}

.drinkInfo {
  margin: 0 auto;
  width: 75%;
}

.drinkContent section {
  align-self: center;
}
