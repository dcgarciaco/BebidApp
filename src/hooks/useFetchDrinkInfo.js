import { useEffect, useState } from 'react';
import { getDrinkInfo } from '../helpers/getDrinkInfo';

export const useFetchDrinkInfo = (drinkID) => {
  const [drinkInfo, setDrinkInfo] = useState({});

  useEffect(() => {
    getDrinkInfo(drinkID).then((drink) => {
      setDrinkInfo(drink);
    });
  }, [drinkID]);

  return drinkInfo;
};
