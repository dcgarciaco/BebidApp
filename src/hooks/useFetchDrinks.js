import { useState, useEffect } from 'react';
import { getDrinks } from '../helpers/getDrinks';

export const useFetchDrinks = (category) => {
  const [state, setstate] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getDrinks(category).then((drink) => {
      setstate({
        data: drink,
        loading: false,
      });
    });
  }, [category]);

  return state;
};
