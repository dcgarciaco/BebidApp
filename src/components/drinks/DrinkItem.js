import './drinkItem.css';
import React from 'react';
import { Redirect, useParams } from 'react-router-dom';

import { useFetchDrinkInfo } from '../../hooks/useFetchDrinkInfo';

export const DrinkItem = () => {
  const { drinkID } = useParams();
  

  const { data: drink, loading } = useFetchDrinkInfo(drinkID);
  if (!drink) {
    return <Redirect to='/' />;
  }
  
  

  const renderIngredientList = (data) => {
  
    const listIngredients = data.map((ingredient, index) => (
      <li key={index}>
        {' '}
        {ingredient[1]} {ingredient[0]}{' '}
      </li>
    ));

    return listIngredients;
  };

  if (loading) {
    return <article className=''>{loading && <p>Loading...</p>}</article>;
  } else {
    return (
      <article className='drinkInfo'>
        <div className='drinkContent'>
          <figure>
            <img src={drink[0].drinkImageURL} alt={drink[0].drinkName} />
          </figure>

          <section>
            <h2>{drink[0].drinkName}</h2>
            <ul>{renderIngredientList(drink[0].drinkIngredients)}</ul>
            <p>
              <b>Glass</b> {drink[0].drinkGlass}
            </p>
            <h3>Instructions</h3>
            <p>{drink[0].drinkInstructions}</p>
          </section>
        </div>
      </article>
    );
  }
};
