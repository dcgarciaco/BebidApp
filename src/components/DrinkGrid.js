import React from 'react';
import { useFetchDrinks } from '../hooks/useFetchDrinks';

import { DrinkGridItem } from './DrinkGridItem';

export const DrinkGrid = ({ category }) => {
  const { data: drinks, loading } = useFetchDrinks(category);

  return (
    <>
      <h3>{category}</h3>

      {loading && <p>Loading</p>}

      <section className='card-grid'>
        {drinks.map((drink) => (
          <DrinkGridItem key={drink.drinkID} {...drink} />
        ))}
      </section>
    </>
  );
};
