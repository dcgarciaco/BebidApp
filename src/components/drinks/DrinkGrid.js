import React from 'react';
import { useFetchDrinks } from '../../hooks/useFetchDrinks';
import { DrinkGridItem } from './DrinkGridItem';
import PropTypes from 'prop-types';

export const DrinkGrid = ({ category }) => {
  const { data: drinks, loading } = useFetchDrinks(category);

  return (
    <>
      {loading && <p>Loading</p>}

      <section className='card-grid'>
        {drinks.map((drink) => (
          <DrinkGridItem key={drink.drinkID} {...drink} />
        ))}
      </section>
    </>
  );
};

DrinkGrid.propType = {
  category: PropTypes.string.isRequired,
};
