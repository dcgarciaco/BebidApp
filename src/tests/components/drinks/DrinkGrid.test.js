import { shallow } from 'enzyme';
import { DrinkGrid } from '../../../components/drinks/DrinkGrid';
import { useFetchDrinks } from '../../../hooks/useFetchDrinks';
jest.mock('../../../hooks/useFetchDrinks');
describe('Test on <DrinkGrid />', () => {
  const category = 'Cocktail';
  test('should show the component correctly', () => {
    useFetchDrinks.mockReturnValue({ data: [], loading: true });
    const wrapper = shallow(<DrinkGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show elements when drinks are loaded from useFetchDrinks', () => {
    const drinks = [
      {
        drinkID: '1111',
        drinkName: 'Test Drink',
        drinkImageURL: 'https://localhost/Test/test.jpg',
      },
      {
        drinkID: '2222',
        drinkName: 'Test Drink',
        drinkImageURL: 'https://localhost/Test/test.jpg',
      },
    ];
    useFetchDrinks.mockReturnValue({ data: drinks, loading: false });
    const wrapper = shallow(<DrinkGrid category={category} />);
  
    expect(wrapper.find('p').exists()).toBeFalsy();
    expect(wrapper.find('DrinkGridItem').length).toBe(drinks.length);
  });
});
