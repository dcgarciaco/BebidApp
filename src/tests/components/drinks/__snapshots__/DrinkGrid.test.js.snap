// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <DrinkGrid /> should show the component correctly 1`] = `
<Fragment>
  <p>
    Loading
  </p>
  <section
    className="card-grid"
  />
</Fragment>
`;
